src/VisualStudio/LiveShare/Impl/CustomProtocol/TextDocumentParams.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Protocol
+{
+    using System.Runtime.Serialization;
+    using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+    /// <summary>
+    /// Class which represents a document reference
+    /// </summary>
+    [DataContract]
+    public class TextDocumentParams
```

Comment:
is this custom protocol, only exist within roslyn?

Reply:
Yes, only exists in Roslyn. That's why I removed it from the Live Share public API.